test/node/node_test.cpp

```diff
@@ -106,6 +130,14 @@ TEST(NodeTest, RemoveUnassignedNode) {
   EXPECT_EQ(0, node.size());
 }
 
+TEST(NodeTest, RemoveUnassignedNodeFromMap) {
```

Comment:
Similarly, please add a tests that verifies that if you have a map, where one of the keys is an integer, you can remove it and it's still a map.